src/PlusDataCollection/PointGrey/vtkPlusSpinnakerVideoSource.cxx

```diff
@@ -0,0 +1,807 @@
+    ptrAcquisitionMode->SetIntValue(acquisitionModeContinuous);
+    LOG_INFO("Acquisition mode set to continuous...");
+
+    // Begin acquiring images
+    this->Internal->CameraPtr->BeginAcquisition();
+
+    std::string deviceSerialNumber("");
+    Spinnaker::GenApi::CStringPtr ptrStringSerial = nodeMapTLDevice.GetNode("DeviceSerialNumber");
+    if (IsAvailable(ptrStringSerial) && IsReadable(ptrStringSerial))
+    {
+      deviceSerialNumber = ptrStringSerial->GetValue();
+      LOG_INFO("Device serial number retrieved as " << deviceSerialNumber << ".");
+    }
+  }
+  catch (Spinnaker::Exception &e)
+  {
+    LOG_ERROR("SpinnakerVideoSource: Failed to start recording. Exception text: " << e.what());
+    return PLUS_FAIL;
+  }
+
+  return PLUS_SUCCESS;
+}
+
+//----------------------------------------------------------------------------
+PlusStatus vtkPlusSpinnakerVideoSource::InternalStopRecording()
+{
+  LOG_TRACE("vtkPlusSpinnakerVideoSource::InternalStopRecording()");
+
+  try
+  {
+    // End acquiring images
+    this->Internal->CameraPtr->EndAcquisition();
+  }
+  catch (Spinnaker::Exception &e)
+  {
+    LOG_ERROR("SpinnakerVideoSource: Failed to end recording. Exception text: " << e.what());
+    return PLUS_FAIL;
+  }
+
+  return PLUS_SUCCESS;
+}
+
+//----------------------------------------------------------------------------
+PlusStatus vtkPlusSpinnakerVideoSource::NotifyConfigured()
+{
+  // TODO: Implement some configuration checks here
+  LOG_TRACE("vtkPlusSpinnakerVideoSource::NotifyConfigured()");
+  return PLUS_SUCCESS;
+}
+
+//----------------------------------------------------------------------------
+PlusStatus vtkPlusSpinnakerVideoSource::InternalUpdate()
+{
+  LOG_TRACE("vtkPlusSpinnakerVideoSource::InternalUpdate()");
+
+  PlusStatus retVal = PLUS_FAIL;
+
+  try
+  {
+    // Retrieve next received image and ensure image completion
+    Spinnaker::ImagePtr pResultImage = this->Internal->CameraPtr->GetNextImage();
+    if (pResultImage->IsIncomplete())
+    {
+      cout << "Image incomplete with image status " << pResultImage->GetImageStatus() << "..." << endl << endl;
+    }
+
+    // Convert image from PointGrey to correct image type
+    Spinnaker::ImagePtr convertedImage;
+    if (this->PixelEncoding == RGB24)
+    {
+      convertedImage = pResultImage->Convert(Spinnaker::PixelFormat_RGB8, Spinnaker::HQ_LINEAR);
+    }
+    else if (this->PixelEncoding == BGR24)
+    {
+      convertedImage = pResultImage->Convert(Spinnaker::PixelFormat_BGR8, Spinnaker::HQ_LINEAR);
+    }
+    else if (this->PixelEncoding == MONO8)
+    {
+      convertedImage = pResultImage->Convert(Spinnaker::PixelFormat_Mono8, Spinnaker::HQ_LINEAR);
+    }
+    else
+    {
+      LOG_ERROR("Invalid image type selected.");
+      return PLUS_FAIL;
+    }
+
+    // add image to PLUS buffer
+
+    // get buffer
+    vtkPlusDataSource* videoSource(NULL);
+    if (this->GetFirstVideoSource(videoSource) != PLUS_SUCCESS)
```

Comment:
Depending on how you set up your expectations in NotifyConfigured you could adjust this code (single or multiple data sources, etc...)